src/WebForms/Compilation/BatchParser.cs

```diff
@@ -52,6 +56,11 @@ internal void Init(VirtualPath virtualPath)
     internal ICollection GetVirtualPathDependencies()
     {
         // Always set the culture to Invariant when parsing (ASURT 99071)
+        if (_isDependenciesCalled)
```

Comment:
couldn't this be done by checking if the _virtualPathDependencies is null? Also, while you're touching it, can you make it a generic collection?

Reply:
the reason is to avoid multiple calls... (we are calling twice at least).. if _virtualPathDependencies is null, it will again parse the header information. So trying to improve the code. Let me know what you think.

Reply:
I meant, couldn't it be guarded with

```cs
if (_virtualPathDependencies is {})
{
  return _virtualPathDependencies;
}

  ... // populate _virtualPathDependencies

  return _virtualPathDependencies;
}
```